presto-main/src/main/java/com/facebook/presto/transaction/InMemoryTransactionManager.java

```diff
@@ -90,8 +90,9 @@
     private final Executor finishingExecutor;
 
     private final Map<String, FunctionNamespaceManager<?>> functionNamespaceManagers = new HashMap<>();
+    private final ImmutableMap<String, String> extraCatalogs;
```

Comment:
nit: `Map<String, String>`. Let's call it `companionCatalogs`. Add a comment to explain that "Some catalog's transaction will be delegated to catalog. Use this enforce both transactions are present".